includes/Media.php

```diff
@@ -273,4 +277,45 @@ public static function upload_mimes( array $mime_types ) {
 		$mime_types['svg'] = 'image/svg+xml';
 		return $mime_types;
 	}
+
+	/**
+	 * Deletes associated poster image when a video is deleted.
+	 *
+	 * This prevents the poster image from becoming an orphan because it is not
+	 * displayed anywhere in WordPress or the story editor.
+	 *
+	 * @param int $attachment_id ID of the attachment to be deleted.
+	 *
+	 * @return void
+	 */
+	public static function delete_video_poster( $attachment_id ) {
+		remove_action( 'pre_get_posts', [ __CLASS__, 'filter_poster_attachments' ] );
+		$query = new WP_Query(
+			[
+				'fields'         => 'ids',
+				'post_status'    => 'any',
+				'post_type'      => 'attachment',
+				'post_parent'    => $attachment_id,
+				// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query'     => [
+					[
+						'key'     => self::POSTER_POST_META_KEY,
+						'compare' => 'EXISTS',
+					],
+				],
+				'no_found_rows'  => true,
+				'posts_per_page' => 1,
+			]
+		);
+		add_action( 'pre_get_posts', [ __CLASS__, 'filter_poster_attachments' ] );
+
+		/**
+		 * Post ID.
+		 *
+		 * @var int $post_id
+		 */
+		foreach ( $query->posts as $post_id ) {
```

Comment:
Could we use `array_map` here?

Reply:
We could, but it would be slower as it's not language construct like `foreach`. Plus, if we do `get_post_meta` here it feels more readable to keep the loop rather than using `array_map` with an anonymous function.